### ISSUE TITLE
Les types Typescript ne sont pas accessibles

### ISSUE THREAD

**Issue by E-OssO** [thecodingmachine/workadventure-map-starter-kit#38]
C'est dommage que les types de votre projets ne soient pas accessible.
Alors que vous y avez apporté beaucoup d'énergie :)

```
let currentPopup: any = undefined;
// Serait mieux avec le bon typage :)
let currentPopup: Popup | undefined = undefined;

```

Important sur les objets manipulable tel que les layers, layersProperties, ...

**Comment by nolway** [comment]
Les types de l'API sont disponibles il suffit d'ajouter ce tag en haut de vos fichiers TS : 
```ts
/// <reference types="@workadventure/iframe-api-typings" />
```

Pour plus d'informations je vous renvois sur la documentation de l'API =) : 
https://workadventu.re/map-building/api-reference

**Comment by E-OssO** [comment]
Effectivement, m'a demande n'était pas assez explicite ^^ il manque les exports de vos types pour qu'on puissent les utiliser.

Oui, ils sont bien défini et accessible. Mais on ne peut pas les utiliser.

Comme j'en avait discuté avec David j'ai vu son issue après avoir remonté celle-ci.
https://github.com/thecodingmachine/workadventure/issues/2313

Par contre, je crois que sa ne couvre pas le type ITiledMapLayer qui est dans mon cas le "principal" dans la manipulation des layers.
Et aussi dans la class ITiledMapLayer le typage de l'attribut properties est surement a améliorer.
Les attributs name, type, value sont tous en any.
Et je pense que properties value mérite son propre type pour éviter d'avoir a énumérer sa liste de types dans chaque déclarations de fonction comme par exemple.

WA.room.setProperty avec la déclaration suivante
setProperty(layerName: string, propertyName: string, propertyValue: string | number | boolean | undefined): void;

**Comment by moufmouf** [comment]
Corrigé par https://github.com/thecodingmachine/workadventure/pull/2319

**Comment by moufmouf** [comment]
Je réouvre, il manque la réexportation de ITiledMapLayer

**Comment by nolway** [comment]
@moufmouf Can you tell me the status of this issue ?
